### PR TITLE
[noScript] Compose

### PR DESCRIPTION
**Pull request by An-dz**
Following #5565 this branch only adds support for **composing** without scripts.

This patch only adds the support but does not automagically make the theme compatible, some changes in the theme must be done. Themes without support still work.

To test this, use my changes in the default theme & composer plugin:
- [nodebb-theme-persona](https://github.com/NodeBB/nodebb-theme-persona/pull/355)
- [nodebb-plugin-composer-default](https://github.com/NodeBB/nodebb-plugin-composer-default/pull/86)

### CODE REVIEW

**Comment by pitaj** [comment]
@An-dz please address the review before we can continue, thanks

**Comment by An-dz** [comment]
This one is complete as well.

File: src/controllers/index.js [hunk 1]
```diff
@@ -279,6 +280,58 @@ Controllers.compose = function (req, res, next) {
...
+				return helpers.noScriptErrors(req, res, err.message, 400);
+			}
+		});
+	} else if (body.cid) {
+		data.cid = body.cid;
+		data.title = body.title;
+		data.tags = [];
+		data.thumb = '';
+
+		async.waterfall([
```
Comment (pitaj) [inline src/controllers/index.js]: I think using async.waterfall here is kinda pointless.

```
topics.post(data, function (err, result) {
  if (err) {
    return helpers.noScriptErrors(req, res, err.message, 400);
  }

  res.redirect(nconf.get('relative_path') + '/topic/' + result.topicData.slug);
});
```

File: src/controllers/index.js [hunk 2]
```diff
@@ -279,6 +280,58 @@ Controllers.compose = function (req, res, next) {
...
+	req.body.noscript = 'true';
+
+	if (!data.content) {
+		return helpers.noScriptErrors(req, res, '[[error:invalid-data]]', 400);
+	}
+
+	if (body.tid) {
+		data.tid = body.tid;
+
+		async.waterfall([
```
Comment (pitaj) [inline src/controllers/index.js]: Same deal up here

File: src/controllers/index.js (hunk 1 above)
Comment (An-dz) [inline src/controllers/index.js]: Yes, you are right, let me test that.

**Comment by An-dz** [comment]
It's updated.

**Comment by An-dz** [comment]
Ok, all done, both scripts and themes are done.

**Comment by julianlam** [comment]
Thanks @An-dz :smile:

**Pull request closed without merging**